huaweicloud/services/cph/data_source_huaweicloud_cph_server_flavors.go

```diff
@@ -0,0 +1,264 @@
+			},
+			"memory": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: `The ram of the CPH server in GB.`,
+			},
+			"flavors": {
+				Type:        schema.TypeList,
+				Elem:        serverFlavorsFlavorsSchema(),
+				Computed:    true,
+				Description: `The list of flavor detail.`,
+			},
+		},
+	}
+}
+
+func serverFlavorsFlavorsSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"flavor_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The name of the flavor.`,
+			},
+			"vcpus": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The vcpus of the CPH server.`,
+			},
+			"memory": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The ram of the CPH server in GB.`,
+			},
+			"type": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The type of the CPH server flavor.`,
+			},
+			"extend_spec": {
+				Type:     schema.TypeList,
+				Elem:     serverFlavorsFlavorsExtendSpecSchema(),
+				Computed: true,
+			},
+		},
+	}
+	return &sc
+}
+
+func serverFlavorsFlavorsExtendSpecSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"vcpus": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The extended description of the vcpus.`,
+			},
+			"memory": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The extended description of the ram.`,
+			},
+			"disk": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The extended description of the disk.`,
+			},
+			"network_interface": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The extended description of the network interface.`,
+			},
+			"gpu": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The extended description of the gpu.`,
+			},
+			"bms_flavor": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The extended description of the bms flavor.`,
+			},
+			"gpu_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The gpu count.`,
+			},
+			"numa_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The numa count.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceServerFlavorsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// listFlavors: Query the list of CPH server flavors
+	var (
+		listFlavorsHttpUrl = "v1/{project_id}/cloud-phone/server-models"
+		listFlavorsProduct = "cph"
+	)
+	listFlavorsClient, err := cfg.NewServiceClient(listFlavorsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating ServerFlavors Client: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("error creating CPH Client: %s", err)
```

can we use the uppercase of `listFlavorsProduct` in template?

Reply:
ok, fix the template